.github/workflows/macos_meson.yml

```diff
@@ -0,0 +1,116 @@
+name: macOS tests (meson)
+
+on:
+  push:
+    branches:
+      - main
+      - maintenance/**
+  pull_request:
+    branches:
+      - main
+      - maintenance/**
+
+env:
+  INSTALLDIR: "installdir"
+  CCACHE_DIR: "${{ github.workspace }}/.ccache"
+
+jobs:
+  test_meson:
+    name: Meson build
+    # If using act to run CI locally the github object does not exist and the usual skipping should not be enforced
+    if: "github.repository != '' || github.repository == 'scipy/scipy' && !contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[skip github]')"
```

Comment:
I find the logic of this line hard to follow. I think the gist is to run the tests if:

- if you're running locally, no matter that the commit message is.
- if you're running on github, skip the test if the commit message contains skip-ci, etc.

Reply:
There are 3 parts:
- allow running locally (with Act)
- disable running on forks by default (that's the `github.repository == 'scipy/scipy`)
- add logic to do the right thing for `[skip github]` and `[skip ci]` in commit messages

I agree this is hard to parse. How about we add a comment explaining the above to all CI jobs?

Reply:
I think I got confused by the associative nature of the booleans, as well as the quotes. I think it's probably mergeable as is. On re-reading it seems a lot clearer.